WoWsShipBuilder.Web/Dialogs/BuildStringInputDialog.razor

```diff
@@ -0,0 +1,108 @@
+﻿@using WoWsShipBuilder.Core.Builds
+@using System.Net
+@inject ILocalizer Localizer
+
+<MudDialog Style="width: 600px">
+    <DialogContent>
+        <MudFocusTrap DefaultFocus="DefaultFocus.FirstChild">
+            <MudTextField @bind-Value="buildString" Label="@Localizer.GetAppLocalization(nameof(Translation.BuildStringInputDialog_EnterBuildString)).Localization" Immediate Validation="@(new Func<string, string?>(ValidateBuildString))"/>
+        </MudFocusTrap>
+    </DialogContent>
+    <DialogActions>
+        <MudButton OnClick="@Cancel">
+            @Localizer.GetAppLocalization(nameof(Translation.Dialog_Cancel)).Localization
+        </MudButton>
+        <MudButton OnClick="@OnConfirm" Disabled="!validBuild">
+            @Localizer.SimpleAppLocalization(nameof(Translation.ShipSelectionWindow_ConfirmButton))
+        </MudButton>
+    </DialogActions>
+</MudDialog>
+
+@code {
+
+    [CascadingParameter]
+    MudDialogInstance MudDialog { get; set; } = new();
+
+    [Parameter]
+    public string SelectedShipIndex { get; set; } = default!;
+    
+    private string buildString = string.Empty;
+
+    private bool validBuild;
+
+    private void Cancel()
+    {
+        MudDialog.Cancel();
+    }
+
+    private void OnConfirm()
+    {
+        string buildStr;
+        if (buildString.Contains("build="))
+        {
+            int index = buildString.IndexOf("build=", StringComparison.Ordinal);
+            buildStr = WebUtility.UrlDecode(buildString[(index + 6)..]);
+        }
+        else
+        {
+            buildStr = buildString;
+        }
+        
+        MudDialog.Close(buildStr.Trim());
+    }
+
+    private string? ValidateBuildString(string buildStr)
+    {
+        validBuild = false;
+        if (string.IsNullOrWhiteSpace(buildStr))
+        {
+            return null;
+        }
+        
+        if (buildStr.Contains("share.wowssb.com"))
```

Comment:
Do not hardcode urls or hostnames, use data from config file.
Also, why is this not supported? You can retrieve the real url by sending a HTTP request with FollowRedirects disabled